packages/babel-plugin-i18n-calypso/src/index.js

```diff
@@ -0,0 +1,202 @@
+					name = callee.property.name;
+				} else {
+					name = callee.name;
+				}
+				if ( 'translate' !== name ) {
+					return;
+				}
+				let i = 0;
+
+				const translation = {
+					msgid: getNodeAsString( path.node.arguments[i++] ),
+					msgstr: '',
+					comments: {},
+				}
+
+				if ( ! translation.msgid.length ) {
+					return;
+				}
+
+				// At this point we assume we'll save data, so initialize if
+				// we haven't already
+				if ( ! baseData ) {
+					baseData = {
+						charset: 'utf-8',
+						headers: state.opts.headers || DEFAULT_HEADERS,
+						translations: {
+							'': {
+								'': {
+									msgid: '',
+									msgstr: [],
+								},
+							},
+						},
+					};
+
+					for ( const key in baseData.headers ) {
+						baseData.translations[ '' ][ '' ].msgstr.push( `${ key }: ${ baseData.headers[ key ] };\n` );
+					}
+
+					// Attempt to exract nplurals from header
+					const pluralsMatch = ( baseData.headers[ 'plural-forms' ] || '' ).match( /nplurals\s*=\s*(\d+);/ );
+					if ( pluralsMatch ) {
+						nplurals = pluralsMatch[ 1 ];
+					}
+				}
+
+				if ( path.node.arguments.length > i ) {
+					const msgid_plural = getNodeAsString( path.node.arguments[i] );
+					if ( msgid_plural.length ) {
+						translation.msgid_plural = msgid_plural;
+						i++;
+						// For plurals, create an empty mgstr array
+						translation.msgstr = Array.from( Array( nplurals ) ).map( () => '' );
+					}
+				}
+
+				const { filename } = this.file.opts;
+				const pathname = relative( '.', filename ).split( sep ).join( '/' );
+				translation.comments.reference = pathname + ':' + path.node.loc.start.line;
+
+				if ( path.node.arguments.length > i && 'ObjectExpression' === path.node.arguments[i].type ) {
+					for ( const j in path.node.arguments[i].properties ) {
+						if ( 'ObjectProperty' === path.node.arguments[i].properties[j].type ) {
+							switch ( path.node.arguments[i].properties[j].key.name ) {
+								case 'context':
+									translation.msgctxt = path.node.arguments[i].properties[j].value.value;
+									break;
+								case 'comment':
+									translation.comments.extracted = path.node.arguments[i].properties[j].value.value;
+									break;
+							}
+						}
+					}
+				}
+
+				// Create context grouping for translation if not yet exists
+				const { msgctxt = '', msgid } = translation;
+				if ( ! strings.hasOwnProperty( msgctxt ) ) {
+					strings[ msgctxt ] = {};
+				}
+
+				if ( ! strings[ msgctxt ].hasOwnProperty( msgid ) ) {
+					strings[ msgctxt ][ msgid ] = translation;
+				} else {
+					strings[ msgctxt ][ msgid ].comments.reference = '\n' + translation.comments.reference;
+				}
+			},
+			Program: {
+				enter() {
+					strings = {};
+				},
+				exit() {
+					if ( isEmpty( strings ) ) {
+						return;
+					}
+
+					const data = merge( {}, baseData, { translations: strings } );
+
+					const compiled = po.compile( data );
+
+					const dir = 'build/.i18n-calypso/';
```

Comment:
I'd rather not hide the dir, since it's just more confusing for devs I guess. Just call it `build/i18n-calypso/` and add it to `.gitignore`?